pkg/image/transport.go

```diff
@@ -129,6 +133,71 @@ func isDir(path string) bool {
 	return strings.HasSuffix(path, "/")
 }
 
+func isFiletypeTar(head []byte) bool {
+	// Signature from https://en.wikipedia.org/wiki/List_of_file_signatures
+	return len(head) > 0x105 &&
+		head[0x101] == 0x75 &&
+		head[0x102] == 0x73 &&
+		head[0x103] == 0x74 &&
+		head[0x104] == 0x61 &&
+		head[0x105] == 0x72
+}
+
+func isFiletypeGzip(head []byte) bool {
+	// Signature from https://en.wikipedia.org/wiki/List_of_file_signatures
+	return len(head) > 1 &&
+		head[0] == 0x1F && head[1] == 0x8B
+}
+
+func guessMediaType(ctx context.Context, src types.ImageSource, layer types.BlobInfo, cache types.BlobInfoCache) (string, error) {
```

Comment:
This feels like it is doing the same as https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/importer/format-readers.go#L227 more or less. But I can't come up with a good way of using that here of the top of my head.

Reply:
Oh, I didn't know about this code.

I've given it some thought. Maybe we could get rid of the whole `MediaType` checking and just use FormatReaders? If a layer can be read using FR, it's good enough for the import process. The declared edia type wouldn't matter.

What do you think?

Reply:
Yeah I think that would be good. I just hate re-implementing code we have been using successfully for some time now. Also maybe some unit tests to verify it is working as expected?